blog/2023-06-22-openfeature-adoption-story.md

```diff
@@ -0,0 +1,130 @@
+Thinking about how to define the exposed interfaces, while keeping it simple and generic, I started doing some research, and encountered [OpenFeature](https://openfeature.dev).
+Gladly, the mentioned feature management cloud service has already created an OpenFeature provider.
+Let me explain why and how it was adopted.
+
+<!--truncate-->
+
+## Why Feature Management was needed
+
+Our product needed a solution for feature flags:
+* providing a gradual release mechanism and a simple way to define target audiences such as target group or split release
+* Using feature flags for dynamic configuration values which can be changed in runtime dynamically without needing for a new release deployment.
+* Ability for A/B testing - can be done via feature for specific region, specific users, etc.
+
+## Why OpenFeature was chosen
+
+While keeping it simple and generic, I had a few wonders:
+* How to name the method
+* How to do provide information for error handling
+* What response structure to define
+* How to define properties passed to the methods
+
+OpenFeature solved me all these questions with its standard. When using a standard, I have good reasons why it is defined like this, as the standard supports it.
+
+## Feature Management Cloud Service Provider Implementation For OpenFeature
+Although the provider implementation is used transparantly, as it is open sourced, I explored its implementation.  
+What I saw is that the provider implementation is very short and simple, acting like a "bridge" from OpenFeature to the feature management cloud service specific vendor SDK which was already exist.  
+This can tell me that the OpenFeature standard is making sense.
+
+## How OpenFeature was adopted
+As for this context OpenFeature can be treated as a "bridge", it was used accordingly:
+* OpenFeature provider initializer was added.
+* Generic OpenFeature client wrapper was created.
+
+## Monitoring
+As our provider was fetching feature flags configuration and receiving real-time configuration changes, we needed to know its status and provide an alert on scenarios like service is down or communication failure.  
+When adoption was done, OpenFeature did not support that functionallity.  
+Open Feature lately added [Events](https://github.com/open-feature/spec/blob/main/specification/sections/05-events.md#5-events). Together with provider events implementation, it could have given us this functionallity.  
+To overcome the missing events enhancement, we implemented it directly via provider abilities. It did cause some mix between OpenFeature and vendor specific implementation.
+
+## Initialization
+A common action is provider initialization. As it was vendor specific, it was done by provider call implementation.  
+Open Feature lately added [Initialization](https://github.com/open-feature/spec/blob/main/specification/sections/02-providers.md#24-initialization) to the spec. This could have make it more standard implementation.
```

Comment:
line 54 change:
-  "lately" to "recently" 
- "make it more" to "made it a more"